### PR TITLE
Allow user to user KCC_USE_DIRECT_RECONCILERS variable

### PR DESCRIPTION
**Pull request by anhdle-sso**
### Change description

Allow user to hotswap KCC_USE_DIRECT_RECONCILERS variable when running test

Test=`export KCC_USE_DIRECT_RECONCILERS=CertificateManagerDNSAuthorization; ./hack/record-gcp fixtures/certificatemanagerdnsauthorization-full`

Fixes #

### Tests you have done

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: dev/tasks/run-e2e [hunk 1]
```diff
@@ -23,8 +23,11 @@ cd ${REPO_ROOT}
 echo "Downloading envtest assets..."
 export KUBEBUILDER_ASSETS=$(go run sigs.k8s.io/controller-runtime/tools/setup-envtest@latest use -p path)
 
-echo "Using direct controllers: SQLInstance,ComputeForwardingRule,GKEHubFeatureMembership"
-export KCC_USE_DIRECT_RECONCILERS=SQLInstance,ComputeForwardingRule,GKEHubFeatureMembership
+if [[ -z "${KCC_USE_DIRECT_RECONCILERS:-}" ]]; then
+  KCC_USE_DIRECT_RECONCILERS=SQLInstance,ComputeForwardingRule,GKEHubFeatureMembership
+fi
+echo "Using direct controllers: $KCC_USE_DIRECT_RECONCILERS"
+export KCC_USE_DIRECT_RECONCILERS
```
Comment (acpana) [inline dev/tasks/run-e2e:30]: Are these defaults? If so, shouldn't the default controller be selected via the annotation (or lack thereof) from the CRD? Or are these resources in transition?

Anyways, I think being able to expose `KCC_USE_DIRECT_RECONCILERS` to the dev tooling makes sense.

File: hack/record-gcp [hunk 2]
```diff
@@ -34,5 +40,5 @@ GOLDEN_REQUEST_CHECKS=1 \
 GOLDEN_OBJECT_CHECKS=1 \
 WRITE_GOLDEN_OUTPUT=1 \
 RUN_E2E=1 \
-KCC_USE_DIRECT_RECONCILERS="SQLInstance,ComputeForwardingRule" \
+KCC_USE_DIRECT_RECONCILERS=$KCC_USE_DIRECT_RECONCILERS \
```
Comment (yuwenma) [inline hack/record-gcp:43]: I think we can remove the `KCC_USE_DIRECT_RECONCILERS` from the `hack/record-gcp` now. @jasonvigil Could you verify the SQLInstance does not need this to run against the real gcp?

Comment (jasonvigil) [reply hack/record-gcp:43]: Yeah let's just remove it. We can now use the `alpha.cnrm.cloud.google.com/reconciler: "direct"` annotation to toggle direct reconciler usage for individual mockgcp test cases. [example](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/f3e33671f78a6fa3b6540bef3e1f8aef44693d51/pkg/test/resourcefixture/testdata/basic/dataflow/v1beta1/dataflowflextemplatejob/batchdataflowflextemplatejob-direct/create.yaml#L20C5-L20C53)

Comment (anhdle-sso) [reply hack/record-gcp:43]: Let me remove `SQLInstance` here

**Comment by acpana** [comment]
From offline conversation, let's keep the `KCC_USE_DIRECT_RECONCILERS` in for now and if and when we remove usage of it we can also remove it from here. Or we can add the annotations on the tests that we care about for `ComputeForwardingRule` w the new annotation as a follow on.

/lgtm
/approve

**Pull request merged**